Core/src/EventData/SpacePointContainer2.cpp

```diff
@@ -148,60 +148,109 @@ void SpacePointContainer2::createExtraColumns(
   if ((columns & R) != None && !m_rColumn.has_value()) {
     m_rColumn = SpacePointExtraColumnHolder<float>();
     m_rColumn->resize(size());
-    m_extraColumns.push_back(&*m_rColumn);
   }
   if ((columns & Phi) != None && !m_phiColumn.has_value()) {
     m_phiColumn = SpacePointExtraColumnHolder<float>();
     m_phiColumn->resize(size());
-    m_extraColumns.push_back(&*m_phiColumn);
   }
   if ((columns & Time) != None && !m_timeColumn.has_value()) {
     m_timeColumn = SpacePointExtraColumnHolder<float>(NoTime);
     m_timeColumn->resize(size());
-    m_extraColumns.push_back(&*m_timeColumn);
   }
   if ((columns & VarianceZ) != None && !m_varianceZColumn.has_value()) {
     m_varianceZColumn = SpacePointExtraColumnHolder<float>();
     m_varianceZColumn->resize(size());
-    m_extraColumns.push_back(&*m_varianceZColumn);
   }
   if ((columns & VarianceR) != None && !m_varianceRColumn.has_value()) {
     m_varianceRColumn = SpacePointExtraColumnHolder<float>();
     m_varianceRColumn->resize(size());
-    m_extraColumns.push_back(&*m_varianceRColumn);
   }
   if ((columns & TopStripVector) != None &&
       !m_topStripVectorColumn.has_value()) {
     m_topStripVectorColumn = SpacePointExtraColumnHolder<Eigen::Vector3f>();
     m_topStripVectorColumn->resize(size());
-    m_extraColumns.push_back(&*m_topStripVectorColumn);
   }
   if ((columns & BottomStripVector) != None &&
       !m_bottomStripVectorColumn.has_value()) {
     m_bottomStripVectorColumn = SpacePointExtraColumnHolder<Eigen::Vector3f>();
     m_bottomStripVectorColumn->resize(size());
-    m_extraColumns.push_back(&*m_bottomStripVectorColumn);
   }
   if ((columns & StripCenterDistance) != None &&
       !m_stripCenterDistanceColumn.has_value()) {
     m_stripCenterDistanceColumn =
         SpacePointExtraColumnHolder<Eigen::Vector3f>();
     m_stripCenterDistanceColumn->resize(size());
-    m_extraColumns.push_back(&*m_stripCenterDistanceColumn);
   }
   if ((columns & TopStripCenter) != None &&
       !m_topStripCenterColumn.has_value()) {
     m_topStripCenterColumn = SpacePointExtraColumnHolder<Eigen::Vector3f>();
     m_topStripCenterColumn->resize(size());
-    m_extraColumns.push_back(&*m_topStripCenterColumn);
   }
   if ((columns & CopyFromIndex) != None && !m_copyFromIndexColumn.has_value()) {
     m_copyFromIndexColumn = SpacePointExtraColumnHolder<std::size_t>();
     m_copyFromIndexColumn->resize(size());
-    m_extraColumns.push_back(&*m_copyFromIndexColumn);
   }
 
   m_knownExtraColumns = m_knownExtraColumns | columns;
+
+  initializeExtraColumns();
+}
+
+void SpacePointContainer2::dropExtraColumns(
+    SpacePointKnownExtraColumn columns) noexcept {
+  using enum SpacePointKnownExtraColumn;
+
+  if ((columns & R) != None) {
```

Comment:
Not only in the diff here, but you could use [`ACTS_CHECK_BIT`](https://github.com/acts-project/acts/blob/main/Core/include/Acts/Utilities/Helpers.hpp#L21) for these expressions to make it more legible. I always have a hard time parsing these binary ops at a glance.

```suggestion
  if (ACTS_CHECK_BIT(columns, R)) {
```

The logic is slightly different, instead of `!= 0` the macro compares with the mask itself, but I think if should give the same result.